pv_site_api/_db_helpers.py

```diff
@@ -112,26 +112,67 @@ def _get_forecasts_for_horizon(
 def _get_latest_forecast_by_sites(
     session: Session,
     site_uuids: list[str],
-    start_utc: Optional[dt.datetime] = None,
+    start_utc: dt.datetime,
     end_utc: Optional[dt.datetime] = None,
     sum_by: Optional[str] = None,
 ) -> list[Row]:
     """Get the latest forecast for given site uuids."""
-    # Get the latest forecast for each site.
-    forecasts = (
-        session.query(ForecastSQL.forecast_uuid)
-        .distinct(ForecastSQL.location_uuid)
-        .filter(ForecastSQL.location_uuid.in_([uuid.UUID(su) for su in site_uuids]))
-        .order_by(
-            ForecastSQL.location_uuid,
-            ForecastSQL.timestamp_utc.desc(),
-        )
-    ).all()
-    forecast_uuids = [forecast.forecast_uuid for forecast in forecasts]
 
     # make conditions and aliases for ML models
     a, b, m_fv, m_site = make_ml_model_alias_and_conditions()
 
+    # find locations, some with ml model attached, some without
+    locations = session.query(LocationSQL).where(LocationSQL.location_uuid.in_(site_uuids)).all()
+
+    location_uuids_wth_ml_models = [
+        loc.location_uuid for loc in locations if loc.ml_model_uuid is not None
+    ]
+    location_uuids_wthout_ml_models = [
+        loc.location_uuid for loc in locations if loc.ml_model_uuid is None
+    ]
+
+    logger.info(
+        f"Getting latest forecast for {len(location_uuids_wth_ml_models)} sites with ML models "
+        f"and {len(location_uuids_wthout_ml_models)} sites without ML models"
+    )
+
+    # Get the latest forecast for each site.
+    # for sites with ml models
+    forecast_uuids = []
+    if len(location_uuids_wth_ml_models) > 0:
+        forecasts = (
+            session.query(ForecastSQL.forecast_uuid)
+            .join(ForecastValueSQL)
+            .distinct(ForecastSQL.location_uuid)
+            .filter(ForecastSQL.location_uuid.in_([su for su in location_uuids_wth_ml_models]))
+            .join(LocationSQL)
```

Comment:
Not quite sure what this query is doing differently – aren't all these query conditions added again below anyway? I'm assuming we're trying to filter by the `ml_model_uuid`, but this is done in the `or(a, b)` condition in both logic branches..?

Reply:
And, if so, not sure what the split is doing – if we need an extra condition adding for when we're filtering/selecting by ML model, can we just conditionally add that condition, if that makes sense?